compiler/tools/fpp-to-cpp/test/component/base/ActiveGetProductsComponentAc.ref.cpp

```diff
@@ -250,7 +250,13 @@ void ActiveGetProductsComponentBase ::
     this->m_cmdIn_InputPort[port].setPortNum(port);
 
 #if FW_OBJECT_NAMES == 1
-    char portName[FW_OBJ_NAME_MAX_SIZE];
+    // The port name consists of this->m_objName and some extra info.
+    // We expect all of this to fit in FW_OBJ_NAME_MAX_SIZE bytes.
+    // However, the compiler may assume that this->m_objName fills
+    // the entire array, whose size is FW_OBJ_NAME_MAX_SIZE. So to
+    // avoid a compiler warning, we provide an extra FW_OBJ_NAME_MAX_SIZE
+    // bytes to cover the extra info.
+    char portName[2*FW_OBJ_NAME_MAX_SIZE];
```

Comment:
The logic holds, per my brief review, as long as `sizeof "_noArgsAsync_InputPort[%" ... "] < FW_OBJ_NAME_MAX_SIZE`.  Otherwise compiler warnings come back....which was an advantage of the previous implementation we saw with the `sizeof` operators.  It ensured enough space for the literals and then let it get cut down to size in the `.setObjName` call.  It also used less stack space when  `sizeof "_noArgsAsync_InputPort[%" ... "] <<< FW_OBJ_NAME_MAX_SIZE`  (by `<<<` I mean much less than).

I don't know if that precision is worth the original complexity...but I can say that it does get annoying when compiler warnings come out by just changing config.

On smaller systems, doubling the stack allocation for a port call here could be significant.  I guess the argument is that object names are not required and as such could be turned off? I can buy that argument.  @timcanham do you agree that turning off FW_OBJECT_NAMES is appropriate for smaller systems?

Reply:
I saw a comment that addresses this in the other PR.  Essentially the argument is that the warning in the case I describe *is valid* because `FW_OBJECT_NAMES` is too small to hold the literal and break.  There are two valid non-warning solutions:

1. Turn off object names
2. Increase object name size.

Reply:
I agree these are the near-term solutions. Tim and I discussed a longer-term solution that allows silent truncation of long object names, and avoids complex coding. See below.